### PR TITLE
docs: Add cryptographic reference documentation.

### PR DESCRIPTION
**Pull request by fnordahl**
It would be useful for our users to be able to look up the details of how MicroOVN makes use of cryptography in its implementation.

### CODE REVIEW

File: docs/reference/cryptography.rst [hunk 1]
```diff
@@ -0,0 +1,28 @@
+============
+Cryptography
+============
+
+Transport Layer Security (TLS)
```
Comment (crypticC0der) [inline docs/reference/cryptography.rst:5]: Shouldnt we be using sentence case for headings?

File: docs/reference/cryptography.rst [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+All network endpoints exposed by MicroOVN services are secured using multiple
+components of the `TLS protocol`_, including encryption, authentication and
+integrity.  Through the use of the `Ubuntu OpenSSL`_ packages, TLS versions
+below 1.2 are disabled for security reasons.
+
+There are two self-signed certificate authorities in use, one for the
+`MicroCluster`_ based ``microovnd`` daemon, another for the `OVN`_ daemons.
+These are initialised during the initial bootstrap of the cluster.
+
+Keys are generated using a 384 bit Elliptic Curve algorithm often referred to
```
Comment (crypticC0der) [inline docs/reference/cryptography.rst:16]: would it be worth linking to further information on this algorithm somewhere?

**Review by crypticC0der — COMMENTED** [review]
Really really good docs
I have some very minor notes, but LGTM

File: docs/reference/cryptography.rst (hunk 1 above)
Comment (fnordahl) [reply docs/reference/cryptography.rst:5]: Ah, indeed, thank you for pointing that out!

File: docs/reference/cryptography.rst (hunk 2 above)
Comment (fnordahl) [reply docs/reference/cryptography.rst:16]: Yeah, we probably should. At first attempt the best reference I found looked a bit outdated, which made me unsure, I think we need something better than Wikipedia.

I'll make another attempt at finding a more recent reference.

Comment (fnordahl) [reply docs/reference/cryptography.rst:16]: It's at least referred to in RFC 5480, let's try that as a reference.

Comment (fnordahl) [reply docs/reference/cryptography.rst]: @crypticC0der pointed to the wikipedia page on [elliptic curve cryptography](https://en.wikipedia.org/wiki/Elliptic-curve_cryptography), while generic it is quite decent, let's use it!

**Review by crypticC0der — APPROVED** [review]
LGTM!

**Pull request merged**